### ISSUE TITLE
Mouse keys enhancement

### ISSUE THREAD

**Issue by jcarlos** [Dygmalab/Bazecor#324]
**Is your feature request related to a problem? Please describe.**
The problem is that you either have a mouse cursor that is too slow or too fast for what you need to do

**Describe the solution you'd like**
Allow to press a secondary key to change mouse speed on demand, ie: let's say I have mouse keys set as ijkl, I define then as fast movement to able to go across the screen fast, however when I'm holding the shift key, the keys ijkl will make the mouse pointer move much slower to I can have higher precision to click on things. 
So basically I want to able to define a key that when pressed or hold will change the mouse speed cursor on demand.

**Describe alternatives you've considered**
Being able to setup multiple mouse keys with different speeds for each so I can implement the describe feature above myself by using layers (one layer has slow mouse, another layer has fast mouse etc..)

**Comment by AlexDygma** [comment]
Hey @jcarlos, great suggestion!

We will go with the "shift the mouse keys" solution for the first option, it will have a toggle in the preferences menu, it should be simple enough to understand by the majority of the users.

**Comment by jcarlos** [comment]
Fantastic, it will be awesome if we can toggle and set the custom speed for the shifted mouse keys in the preferences menu.

**Comment by jkingsummit** [comment]
@AlexDygma what is the status of this feature?

**Comment by jkingsummit** [comment]
Also, incidental finding, I was trying to accomplish something as a 'stop-gap' measure and noticed that it seems mouse movement isn't working in macros

**Comment by AlexDygma** [comment]
Hey there @jkingsummit and @jcarlos !

Right now the 4 Dygma firmware engineers are all fully focused on the Defy, so these tasks have stalled and will probably remain so until January 😓, time on which we will be building the manufacturing testing equipment, but will have more time to update new features of the Defy

**Comment by HanssenKai** [comment]
Instead of having it in the preferences menu, perhaps it would be better to implement a new row to the "mouse events" dropdown in the keys selector e.g "mouse **speed modifier**" with a slider starting at 0% -+200%

**Comment by sonorousfreq** [comment]
Hi,

any updates on this?

The mouse acceleration and speed also behaves strangely on my Dygma, it changes the speed somehow when cursor speed slider in preferences is touched, then changing cursor acceleration makes it much faster, and again changin cursor speed (either way) makes it slow again, like one setting is overriding the other. Are there any tutorials on how to set this up?

The speed also varies between OSes (win/unix).

**Comment by MiquelDygma** [comment]
Hi!

Speed may vary between OSes because it's relative to your system pointer speed. 

On the other hand, there's a bug where if you only modify one mouse speed setting, the rest will be reset. To avoid that, you need to modify them all at once (even if just by 1 unit).
